employee-rostering-frontend/src/ui/pages/rotation/RotationPage.tsx

```diff
@@ -0,0 +1,369 @@
+      return prev.set(curr.spot.id as number, old.concat(templatesToAdd));
+    },
+    spotSelectors.getSpotList(state).reduce((prev, curr) => prev.set(curr.id as number, []),
+      new Map<number, ShiftTemplate[]>())),
+  rosterState: state.rosterState.rosterState
+}); 
+  
+export interface DispatchProps {
+  addShiftTemplate: typeof shiftTemplateOperations.addShiftTemplate;
+  removeShiftTemplate: typeof shiftTemplateOperations.removeShiftTemplate;
+  updateShiftTemplate: typeof shiftTemplateOperations.updateShiftTemplate;
+  showInfoMessage: typeof alert.showInfoMessage;
+}
+  
+const mapDispatchToProps: DispatchProps = {
+  addShiftTemplate: shiftTemplateOperations.addShiftTemplate,
+  removeShiftTemplate: shiftTemplateOperations.removeShiftTemplate,
+  updateShiftTemplate: shiftTemplateOperations.updateShiftTemplate,
+  showInfoMessage: alert.showInfoMessage
+}
+  
+export type Props = StateProps & DispatchProps;
+interface State {
+  shownSpot: Spot|null;
+  isCreatingOrEditingShiftTemplate: boolean;
+  selectedShiftTemplate?: ShiftTemplate;
+  weekNumber: number;
+}
+
+export const baseDate = moment("2018-01-01T00:00").startOf('week').toDate();
+
+const ShiftTemplatePopoverHeader: React.FC<{
+  shiftTemplate: ShiftTemplate;
+  rotationLength: number;
+  onEdit: (shift: ShiftTemplate) => void;
+  onDelete: (shift: ShiftTemplate) => void;
+}> = props =>{
+  const { t } = useTranslation();
+  const durationBetweenRotationStartAndEnd = moment
+    .duration(props.shiftTemplate.durationBetweenRotationStartAndTemplateStart)
+    .add(props.shiftTemplate.shiftTemplateDuration);
+  return (
+    <span> 
+      <Text> 
+        {t("shiftTemplate", {
+          spot: props.shiftTemplate.spot.name,
+          rotationEmployee: props.shiftTemplate.rotationEmployee? props.shiftTemplate.rotationEmployee.name
+            : "Unassigned",
+          dayStart: Math.floor(modulo(
+            props.shiftTemplate.durationBetweenRotationStartAndTemplateStart.asDays(),
+            props.rotationLength)) + 1,
+          startTime: moment("2018-01-01")
+            .add(props.shiftTemplate.durationBetweenRotationStartAndTemplateStart).format("LT"),
+          dayEnd: Math.floor(modulo(durationBetweenRotationStartAndEnd.asDays(),
+            props.rotationLength)) + 1,
+          endTime: moment("2018-01-01")
+            .add(durationBetweenRotationStartAndEnd).format("LT")
+        })}
+
+      </Text>
+      <Button
+        onClick={() => props.onEdit(props.shiftTemplate)}
+        variant={ButtonVariant.link}
+      >
+        <EditIcon />
+      </Button>
+      <Button
+        onClick={() => props.onDelete(props.shiftTemplate)}
+        variant={ButtonVariant.link}
+      >
+        <TrashIcon />
+      </Button>
+    </span>
+  )}
+
+const ShiftTemplatePopoverBody: React.FC = () => <></>;
+
+const ShiftTemplateEvent: React.FC<EventProps<ShiftTemplate>> = (props) => {
+
+  return (
+    <span
+      data-tip
+      data-for={String(props.event.id)}
```

Comment:
How do you consume these attributes?

Reply:
IIRC, these attributes are consumed by the popup; however, I might have changed the popup technology (so this may not be needed anymore)

Reply:
Yes, please check if it's still needed. I grepped the code and didn't find any occurrences that would refer to this so it really looks like it's unused. If it's consumed by a third party library please point me to that. It might be worth leaving a comment explaining what the attribute does.

Reply:
Not needed; removed.